astropy/io/registry.py

```diff
@@ -599,13 +607,46 @@ def _get_valid_format(mode, cls, path, fileobj, args, kwargs):
                               "The available formats are:\n"
                               "{}".format(format_table_str))
     elif len(valid_formats) > 1:
-        raise IORegistryError(
-            "Format is ambiguous - options are: {}".format(
-                ', '.join(sorted(valid_formats, key=itemgetter(0)))))
+        return _get_highest_priority_format(mode, cls, valid_formats)
 
     return valid_formats[0]
 
 
+def _get_highest_priority_format(mode, cls, valid_formats):
+    """
+    Returns the reader or writer with the highest priority. If it is a tie,
+    error.
+    """
+    if mode == "read":
+        format_dict = _readers
+        mode_loader = "reader"
+    elif mode == "write":
+        format_dict = _writers
+        mode_loader = "writer"
+
+    best_formats = []
+    current_priority = - np.inf
+    for format in valid_formats:
+        try:
+            _, priority = format_dict[(format, cls)]
+        except KeyError:
+            # We could throw an exception here, but get_reader/get_writer handle
+            # this case better, instead maximally deprioritise the format.
+            priority = - np.inf
+
+        if priority == current_priority:
+            best_formats.append(format)
+        elif priority > current_priority:
+            best_formats = [format]
+            current_priority = priority
+
+    if len(best_formats) == 1:
+        return best_formats[0]
+    raise IORegistryError(
+        "Format ambiguous, priorities are tied - best formats are: {}".format(
```

Comment:
This exception message is basically giving the same idea as the previous one, but the different choice of words will break tests for other packages that might be defining their own readers. So I'd suggest just emitting the same exception message as before. (I.e. - whenever a change "breaks" existing astropy tests then you have to think about whether it will break other package's tests).

Reply:
Sure, I've switched back to the old text.